aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -915,4 +922,41 @@ static dispatch_data_t getSectionData(const std::string& name) {
   return l;
 }
 
+// MetalKernelFunction implementation
+MetalKernelFunction::~MetalKernelFunction() {}
```

Comment:
```suggestion
MetalKernelFunction::~MetalKernelFunction() = default;
```
Or is that not supported in this context?

Reply:
I'm still debating on whether to delete the destructor, or actually free the resource, which is borrowed from the MetalShaderLibrary